File: Assemblies/Protean.Tools.vb/Integration/Twitter/TwitterVB2/Objects/PagedResults.vb
Namespace Integration.Twitter.TwitterVB2
    Public Class PagedResults(Of T)
        Inherits List(Of T)

        Public Sub New()
            LCursor = -1
        End Sub

        Public Sub New(ByVal Cursor As Long)
            LCursor = Cursor
        End Sub

        Dim LCursor As Long
        Public Property Cursor() As Long
            Get
                Return LCursor
            End Get
            Set(ByVal value As Long)
                LCursor = Value
            End Set
        End Property

        Public ReadOnly Property HasMore() As Boolean
            Get
                Return Cursor <> 0
            End Get
        End Property

    End Class
End Namespace

File: Assemblies/Protean.Tools_Core/Integration/Twitter/TwitterVB2/Objects/PagedResults.cs
using System.Collections.Generic;
namespace Protean.Tools.Integration.Twitter.TwitterVB2
{
    public partial class PagedResults<T> : List<T>
    {

        public PagedResults()
        {
            LCursor = -1;
        }

        public PagedResults(long Cursor)
        {
            LCursor = Cursor;
        }

        private long LCursor;
        public long Cursor
        {
            get
            {
                return LCursor;
            }
            set
            {
                LCursor = value;
            }
        }

        public bool HasMore
        {
            get
            {
                return Cursor != 0L;
            }
        }

    }
}
